tip-distributor/src/claim_mev_workflow.rs

```diff
@@ -30,123 +36,408 @@ pub enum ClaimMevError {
-            let account = rpc_client.get_account(&tree.tip_distribution_account).await.expect("expected to fetch tip distribution account");
-            let fetched_tip_distribution_account = TipDistributionAccount::try_deserialize(&mut account.data.as_slice()).expect("failed to deserialize tip_distribution_account state");
-            if fetched_tip_distribution_account.merkle_root.is_none() {
-                info!(
-                        "not claiming because merkle root isn't uploaded yet. skipped {} claimants for tda: {:?}",
-                        tree.tree_nodes.len(),
-                        tree.tip_distribution_account
-                    );
+        let transactions_len = transactions.len();
+
+        info!("Sending {} tip claim transactions. {zero_lamports_count} would transfer zero lamports, {below_min_rent_count} would be below minimum rent", transactions.len());
+        let send_start = Instant::now();
+        let (remaining_transaction_count, new_failed_transaction_count) =
+            sign_and_send_transactions_with_retries_multi_rpc(
+                &keypair,
+                &blockhash_rpc_client,
+                &rpc_clients,
+                transactions,
+                max_loop_duration,
+            )
+            .await;
+        failed_transaction_count =
+            failed_transaction_count.saturating_add(new_failed_transaction_count);
+
+        datapoint_info!(
+            "claim_mev_workflow-send_transactions",
+            ("epoch", merkle_trees.epoch, i64),
+            ("attempt", retries, i64),
+            ("transaction_count", transactions_len, i64),
+            (
+                "successful_transaction_count",
+                transactions_len.saturating_sub(remaining_transaction_count),
+                i64
+            ),
+            (
+                "remaining_transaction_count",
+                remaining_transaction_count,
+                i64
+            ),
+            (
+                "failed_transaction_count",
+                new_failed_transaction_count,
+                i64
+            ),
+            ("send_latency_us", send_start.elapsed().as_micros(), i64),
+        );
+
+        if retries >= max_loop_retries {
+            return Err(ClaimMevError::MaxSendTransactionRetriesExceeded {
+                attempts: max_loop_retries,
+                remaining_transaction_count,
+                failed_transaction_count,
+            });
+        }
+        retries = retries.saturating_add(1);
+    }
+}
+
+#[allow(clippy::result_large_err)]
+fn build_transactions(
+    tip_distribution_program_id: &Pubkey,
+    merkle_trees: &GeneratedMerkleTreeCollection,
+    payer_pubkey: &Pubkey,
+    tree_nodes: &[&TreeNode],
+    minimum_rent: u64,
+    tdas: &HashMap<Pubkey, TipDistributionAccount>,
+    claimants: &HashMap<Pubkey, u64>,
+    claim_statuses: &HashMap<Pubkey, Option<Account>>,
+) -> Result<(usize, usize, usize, usize, Vec<Transaction>), ClaimMevError> {
```

Comment:
can you add comments what each `usize` is?

```
-> Result<(usize /* thing */, ..), ClaimMevError>
```

Reply:
fixed, ty